src/frontend/context/MapStyleContext.tsx

```diff
@@ -25,16 +47,52 @@ export const MapStyleContext: React.Context<MapStyleContextType> = React.createC
 const usePersistedState = createPersistedState(MAP_STYLE_KEY);
 
 export const MapStyleProvider: React.FC = ({ children }) => {
-  const [styleId, , setStyleId] = usePersistedState<string | null>(null);
+  const [mapServerReady, setMapServerReady] = React.useState(false);
+  const [styleId, status, setStyleId] = usePersistedState<string>("");
+  const [{ backgroundMaps }] = useExperiments();
+
+  const [onlineMapState, setOnlineState] = React.useState<OnlineState>(
+    "unknown"
+  );
 
-  const contextValue = React.useMemo(() => [styleId, setStyleId] as const, [
-    styleId,
-    setStyleId,
-  ]);
+  // TODO: Eventually use the net info module to determine if internet access is available
+  // Currently not truly a reflection of internet access
+  React.useEffect(() => {
+    let didCancel = false;
+
+    ky.get(normalizeStyleURL(onlineStyleURL, config.mapboxAccessToken))
+      .json()
+      .then(() => didCancel || setOnlineState("online"))
+      .catch(() => didCancel || setOnlineState("offline"));
+
+    return () => {
+      didCancel = true;
+    };
+  }, []);
```

Comment:
I'm fixing this now, but just wanted to drop a note somewhere about it: previously this would run every time the user returned to the map screen, to check if the user was online. This change (moving it into context) means that it only runs when the app first opens, so if the user moves offline then this will never change. As mentioned, ideally we use a net info module to detect an online/offline state change, or we poll, so that we can detect a change to offline state even without navigating away and back to the map.